stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -211,6 +216,69 @@ public static <ReqT, RespT> Iterator<RespT> blockingServerStreamingCall(
     return result;
   }
 
+  /**
+   * The better way to do server streaming calls.<br>
+   * Call {@link BlockingClientCall#read()} for
+   * retrieving values.  A {@code null} will be returned after the server has closed the stream.
+   * <br>
+   * The methods {@link BlockingClientCall#hasNext()} and {@link
+   * BlockingClientCall#cancel(String, Throwable)} can be used for more extensive control.
+   * <br>
+   * <p> Example usage:</p>
+   * <pre>
+   * {@code  while ((response = call.read()) != null) { ... } }
+   * </pre>
+   * or
+   * <pre>
+   * {@code
+   *   while (call.hasNext()) {
+   *     response = call.read();
+   *     ...
+   *   }
+   * }
+   * </pre>
+   *
+   * @return A {@link BlockingClientCall} that has had the request sent and halfClose called
+   * @throws InterruptedException if it receives an interrupt while sending the request
+   */
+  public static <ReqT, RespT> BlockingClientCall<?, RespT> blockingV2ServerStreamingCall(
+      Channel channel, MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, ReqT req)
+      throws InterruptedException {
+    io.grpc.stub.BlockingClientCall<ReqT, RespT> call =
+        blockingBidiStreamingCall(channel, method, callOptions);
+    call.write(req);
+    call.halfClose();
+    return call;
+  }
+
+  public static <ReqT, RespT> BlockingClientCall<ReqT, RespT> blockingClientStreamingCall(
+      Channel channel, MethodDescriptor<ReqT, RespT> method, CallOptions callOptions) {
+    return blockingBidiStreamingCall(channel, method, callOptions);
+  }
+
+  /**
+   * Initiate a bidirectional-streaming {@link ClientCall} and returning a stream object
+   * ({@link BlockingClientCall}) which can be used by the client to send and receive messages over
+   * the grpc channel.
+   *
+   * @return an object representing the call which can be used to read, write and terminate it.
+   */
+  public static <ReqT, RespT> BlockingClientCall<ReqT, RespT> blockingBidiStreamingCall(
+      Channel channel, MethodDescriptor<ReqT, RespT> method, CallOptions callOptions) {
+    ThreadSafeThreadlessExecutor executor = new ThreadSafeThreadlessExecutor();
+    ClientCall<ReqT, RespT> call = channel.newCall(method,
+        callOptions.withOption(ClientCalls.STUB_TYPE_OPTION, StubType.BLOCKING)
+            .withExecutor(executor));
+
+    BlockingClientCall<ReqT, RespT> blockingClientCall = new BlockingClientCall<>(call, executor);
+
+    // Get the call started
+    call.start(blockingClientCall.getListener(), new Metadata());
+    call.request(1);
```

Comment:
We should double-check whether we can avoid the implicit read. I know the details were subtle, but if we are able to avoid the request(1) here, then the only reason to choose between blocking and async becomes threading model (vs features available).